lesson_source/0024/pl.md

```diff
@@ -0,0 +1,170 @@
+
+Wycofaj ostatnią zapisaną zmianę (commit)
+
+    :::bash
+    git revert HEAD
+
+
+Konfiguracja GitHuba - klucze SSH
+---------------------------------
+
+Wygeneruj klucz SSH
+
+    :::bash
+    ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+
+Skopiuj klucz do schowka
+
+    :::bash
+    clip < ~/.ssh/id_rsa.pub
+
+[Dodaj klucz do konta na GitHub](https://ipatryk.pl/polaczenie-do-github-przez-ssh/)
+
+
+Zdalne repozytorium
+---------------------------------
+
+Podepnij zdalne repozytorium do repozytorium lokalnego
+
+    :::bash
+    git remote add origin git@github.com:/user/repo.git
+
+Sprawdź adres zdalnego repozytorium
+
+    :::bash
+    git remote -v
+
+Zmień adres zdalnego repozytorium
+
+    :::bash
+    git remote set-url origin git@github.com:/user/repo.git
+
+Dodaj zmiany zapisane lokalnie do zdalnego repozytorium
+
+    :::bash
+    git push
+
+Swórz kopię lokalną zdalnego repozytorium
+
+    :::bash
+    git clone git@github.com:user/repo.git
+
+[Dodawanie repozytorium zdalnego](https://help.github.com/articles/adding-a-remote/)
+[Zmiana adresu zdalnego repozytorium](https://help.github.com/articles/changing-a-remote-s-url/)
+
+
+Najważniejsze pojęcia
+---------------------------------
+**Branch (Gałąź)** - To Twoja bezpieczna strefa do eksperymentów. Możesz na niej swobodnie wprowadzać zmiany, bez obaw, że popsujesz działającą już wersję kodu. Gałęzie tworzymy za pomocą polecenia `git checkout -b <nazwa_gałęzi>`. Po zakończeniu pracy nad nowymi zmianami i zapisaniu ich, możesz scalić swoją eksperymentalną gałąź ze źródłem projektu przechodząc do niego przez polecenie `git checkout master`, a następnie wywołując polecenie `git merge <nazwa_gałęzi>`.
+
+**Clone** - Lokalna kopia zdalnego repozytorium. Po utworzeniu forka, możemy zapisać skopiowane repozytorium lokalnie, za pomocą polecenia `git clone <adres_naszego_forka>` . Dzięki temu, możemy pracować na tym repozytorium tak samo, jak na każdym innym stworzonym przez nas lokalnym repozytorium.
+
+**Commit** - Zapisanie stanu projektu w historii wersji. Tworzy się je za pomocą polecenia `git commit -m <krótki opis>`. Każdy commit stanowi swoistą migawkę stanu projektu do którego możesz zawsze wrócić. Listę dotychczas zapisanych commitów możesz przeglądać wpisując polecenie `git log`.
+
+**Fork** - Zdalna kopia innego zdalnego repozytorium. Forka tworzymy gdy chcemy na swoim koncie GitHub zapisać kopię czyjegoś repozytorium również dostępnego na GitHub. Forki tworzymy poprzez GitHuba, wchodząc na adres repozytorium, które chcemy skopiować i klikając odpowiednią ikonkę.
+
+**Git Repository** - Repozytorium to w zasadzie folder zawierający wszystkie pliki jakie chcesz śledzić za pomocą Gita. Zwykle tworzy się jedno repozytorium per projekt. Repozytoria lokalne tworzymy za pomocą polecenia `git init .`.
+
+**Master (Źródło)** - To rdzeń Twojego projektu, który zawiera najświeższą, działającą wersję kodu. W źródle zapisujemy kompletne, nowe wersje kodu.
```

Comment:
master i orgian to mogą być dwie różne rzeczy
master domyslny branch w repozytorium do którego nie powinny sie bezpośrednio commitować
orgin (-master) w przeciwieństwie do remote(-master) jest tylko jeden może być lokalny lub zdalny wykorzysutje się w modelu fork-PR szczególnie jak jest potrzeba pracy na czyimś forku jest to istotne.

" który zawiera najświeższą, działającą wersję kodu." powinine zawierać ;) 
to jest punkt wyjść dla osób które chcą dołączyć do projektu

Reply:
sorry za mój polski